plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -768,6 +775,25 @@ private static SliceWriteFunction nvarcharWriteFunction()
                 .flatMap(dataCompression -> Enums.getIfPresent(DataCompression.class, dataCompression).toJavaUtil());
     }
 
+    private static Optional<DataCompression> getTableDataCompressionWithRetries(Handle handle, JdbcTableHandle table)
+    {
+        // DDL operations can take out locks against system tables causing the `getTableDataCompression` query to deadlock
+        // Retry if this deadlock occurs (SQLServerError error number 1205)
+        RetryPolicy<Optional<DataCompression>> retryPolicy = new RetryPolicy<Optional<DataCompression>>()
+                .withMaxAttempts(3)
+                .handleIf(throwable ->
+                {
+                    Throwable rootCause = Throwables.getRootCause(throwable);
+                    return rootCause instanceof SQLServerException &&
+                            ((SQLServerException)(rootCause)).getSQLServerError().getErrorNumber() == 1205;
+                })
+                .onFailedAttempt(event -> log.warn(event.getLastFailure(), "Error when getting table compression info for '%s'", table));
```

Comment:
nit: the message makes it look a permanent failure, it should mention that a retry is being done, ideally even the count of current attempt?